lms/static/js/discovery/discovery_factory.js

```diff
@@ -30,9 +30,12 @@
             }
             listing = new CoursesListing({model: courseListingModel});
 
-            dispatcher.listenTo(form, 'search', function(query) {
-                form.showLoadingIndicator();
-                search.performSearch(query, filters.getTerms());
+            dispatcher.listenTo(form, "search", function (query) {
+            form.showLoadingIndicator();
+            if (!query || query.trim() === "") {
+                filters.remove("search_query");
+            }
+            search.performSearch(query, filters.getTerms());
             });
```

Comment:
```suggestion
            dispatcher.listenTo(form, "search", function (query) {
                form.showLoadingIndicator();
                if (!query || query.trim() === "") {
                    filters.remove("search_query");
                }
                search.performSearch(query, filters.getTerms());
            });
```

Let's fix the spacing so this reads a bit better.